### ISSUE TITLE
Segfault on topkg install

### ISSUE THREAD

**Issue by kayceesrk** [ocamllabs/ocaml-multicore#78]
`opam install topkg` segfaults. The version number is 0.8.1. Here is the error log from `ocamlrund`:

```
kc@ubuntu:~/.opam/4.02.2+local-git-master/build/topkg.0.8.1$ ocamlrund ocaml pkg/pkg.ml build --pkg-name topkg --pinned false
[-1] Opening bytecode executable /home/kc/.opam/4.02.2+local-git-master/bin/ocamlrund
[-1] Not a bytecode executable
[-1] Opening bytecode executable /home/kc/.opam/4.02.2+local-git-master/bin/ocamlrund
[-1] Not a bytecode executable
[-1] Opening bytecode executable /home/kc/.opam/4.02.2+local-git-master/bin/ocaml
[-1] Initial stack limit: 8192k bytes
[00] caml_promote: forcing minor GC. %_minor_to_scan=24.574661
[00] Minor collection of domain 0 starting
[00] Mark stack overflow. Postponing 2 pools (24.6%, leaving 12349).
[00] Mark stack overflow. Postponing 3 pools (26.9%, leaving 11978).
[00] Minor collection of domain 0 completed: 99% of 505 KB live, 26085 pointers rewritten
[00] Minor collection of domain 0 starting
[00] Mark stack overflow. Postponing 3 pools (25.0%, leaving 12292).
[00] Mark stack overflow. Postponing 2 pools (20.6%, leaving 13013).
[00] Minor collection of domain 0 completed: 36% of 2048 KB live, 10463 pointers rewritten
[00] Pooled memory: 286720 alloced, 97902 free, 1204 fragmentation
[00] Large memory: 56479 alloced, 0 free, 62 fragmentation
[00] Redarkening pool 0x7f26dc3a8000 (9 others left)
[00] Redarkening pool 0x7f26dc448000 (8 others left)
[00] Redarkening pool 0x7f26dd528000 (7 others left)
[00] Redarkening pool 0x7f26dc438000 (6 others left)
[00] Redarkening pool 0x7f26dc430000 (5 others left)
[00] Redarkening pool 0x7f26dc440000 (4 others left)
[00] Redarkening pool 0x7f26dc458000 (3 others left)
[00] Redarkening pool 0x7f26dc470000 (2 others left)
[00] Redarkening pool 0x7f26dc468000 (1 others left)
[00] Redarkening pool 0x7f26dd530000 (0 others left)
[00] Major slice: 243751 alloc, 1279692 work, 0 sweep, 214036 mark (53085 blocks)
[00] Contributing 256 heaps to GC
[00] Pooled memory: 286720 alloced, 97902 free, 1204 fragmentation
[00] Large memory: 56479 alloced, 0 free, 62 fragmentation
[00] Minor collection of domain 0: skipping
[00] Finished marking major heap. Marked 53085 blocks
[00] GC cycle completed (heap cycled)
[00] Cycling heap [00]
[00] Minor collection of domain 0: skipping
[00] Growing global data to 5632 entries
Segmentation fault (core dumped)
```

**Comment by kayceesrk** [comment]
This turns out to be a hilarious bug. Noticed that just before segfaulting, the program calls into `Gc.get()`. Now, why should anyone call into asking for GC stats other than to perf test? `topkg` loads `topfind` which uses this abomination to determine whether the program is running in native or bytecode mode: https://gitlab.camlcity.org/gerd/lib-findlib/blob/master/src/findlib/topfind_rd1.p#L26-28. Since multicore doesn't implement `Gc.get` yet and always returns `Val_unit`: https://github.com/ocamllabs/ocaml-multicore/blob/master/byterun/gc_ctrl.c#L329, the code natually segfaults. 

There is a better way to query the backend type, https://github.com/ocaml/ocaml/pull/473, but this is only available in 4.04. For now, my plan is to match the behaviour for stack limit to get `findlib` to work.

**Comment by kayceesrk** [comment]
Fixed by https://github.com/ocamllabs/ocaml-multicore/commit/0b2e9d796fbacd223772c47c86f6874a1d506828.
